File: spec/javascript/test_floating_points.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('FloatingPoints', 'src/floating_points.bin', function(r) {
    var delta = 1e-6;
    
    assert(Math.abs(r.singleValue - 1.2345) < delta);
    assert(Math.abs(r.singleValueBe - 1.2345) < delta);
    assert(Math.abs(r.doubleValue - 123.456) < delta);
    assert(Math.abs(r.doubleValueBe - 123.456) < delta);
    
    assert(Math.abs(r.singleValuePlusInt - 2.2345) < delta);
    assert(Math.abs(r.singleValuePlusFloat - 1.7345) < delta);
    assert(Math.abs(r.doubleValuePlusFloat - 123.506) < delta);
});

File: spec/lua/test_floating_points.lua
local luaunit = require("luaunit")

require("floating_points")

TestFloatingPoints = {}

function TestFloatingPoints:test_floating_points()
    local r = FloatingPoints:from_file("src/floating_points.bin")
    local prec = 0.000001

    luaunit.assertEquals(r.single_value, 0.5)
    luaunit.assertEquals(r.single_value_be, 0.5)
    luaunit.assertEquals(r.double_value, 0.25)
    luaunit.assertEquals(r.double_value_be, 0.25)

    luaunit.assertAlmostEquals(r.approximate_value, 1.2345, prec)

    luaunit.assertAlmostEquals(r.single_value_plus_int, 1.5, prec)
    luaunit.assertAlmostEquals(r.single_value_plus_float, 1.0, prec)
    luaunit.assertAlmostEquals(r.double_value_plus_float, 0.3, prec)
end
